### PR TITLE
ZSH support

### LINKED ISSUE #18: ZSH support
Support spawning ZSH sub shells.

As mentioned in the recent [blog post](https://blog.sbstp.ca/introducing-kubie/), zsh support is planned for the future.

### PR DESCRIPTION
**Pull request by sbstp**
Fixes #18

### CODE REVIEW

**Comment by pxtxs** [comment]
What do you think about creating a crate for the detection module ? Instead of implementing the code directly in kubie. Could be useful.

**Comment by sbstp** [comment]
Could be useful indeed. However I don't really want to add support for Windows and more shells. If I publish it on crates.io people might bother me to add support for more shells.

File: src/shell/mod.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    prompt: String,
+}
+
+pub fn spawn_shell(settings: &Settings, config: KubeConfig, session: &Session) -> Result<()> {
+    let kind = match &settings.shell {
+        Some(shell) => ShellKind::from_str(&shell).ok_or_else(|| anyhow!("Invalid shell setting: {}", shell))?,
+        None => detect()?,
+    };
+
+    let temp_config_file = Tempfile::new("/tmp", "kubie-config", ".yaml")?;
```
Comment (sbstp) [inline src/shell/mod.rs]: Remove custom tempfile module, rely on tempfile crate.

File: src/shell/detect.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+/// This function walks up the process tree and finds all the ancestors to kubie.
+/// If any of kubie's ancestor is a known shell, we have found which shell is in
+/// use.
+///
+/// This functions depends on the `ps` command being installed and available in
+/// the PATH variable.
+///
+/// The SHELL environment variable corresponds to the user's configured SHELL, not
+/// the shell currently in use.
+pub fn detect() -> Result<ShellKind> {
```
Comment (sbstp) [inline src/shell/detect.rs:72]: Detection does not work on OS X, needs to be investigated. Different ps behavior?

**Pull request merged**